node/rpc/src/lib.rs

```diff
@@ -51,114 +72,331 @@ where
+                                        code: 1,
+                                        message: "Error serialization request",
+                                    },
+                                    id: Id::Null,
+                                });
+                                streaming_payload_and_log(
+                                    ws_sender.clone(),
+                                    response,
+                                    subscriber.clone(),
+                                    chain_notify_count,
+                                )
+                                .await;
+                            }
+                        }
+                    }
+                    Err(_) => {
+                        let response = ResponseObjects::One(ResponseObject::Error {
+                            jsonrpc: V2,
+                            error: Error::Provided {
+                                code: 2,
+                                message: "Error reading request",
+                            },
+                            id: Id::Null,
+                        });
+                        streaming_payload_and_log(
+                            ws_sender.clone(),
+                            response,
+                            subscriber.clone(),
+                            chain_notify_count,
+                        )
+                        .await;
+                    }
+                };
+            }
+        } else {
+            warn!("web socket connection failed at {:}", addr)
+        }
+    })
+}
+
+async fn streaming_payload_and_log(
+    ws_sender: Arc<RwLock<WsSink>>,
+    response_object: ResponseObjects,
+    mut subscriber: Subscriber<IndexToHeadChangeJson>,
+    streaming_count: usize,
+) {
+    let response_text = serde_json::to_string(&response_object).unwrap();
+    ws_sender
+        .write()
+        .await
+        .send(Message::text(response_text))
+        .await
+        .unwrap_or_else(|_| warn!("Could not send to response to socket"));
+    if let ResponseObjects::One(ResponseObject::Result {
+        jsonrpc: _,
+        result: _,
+        id: _,
+        streaming,
+    }) = response_object
+    {
+        if streaming {
+            task::spawn(async move {
+                let sender = ws_sender.clone();
+                while let Some(index_to_head_change) = subscriber.next().await {
+                    if streaming_count == index_to_head_change.0 {
+                        let head_change = (&index_to_head_change.1).into();
+                        let data = StreamingData {
+                            json_rpc: "2.0".to_string(),
+                            method: "xrpc.ch.val".to_string(),
+                            params: (streaming_count, vec![head_change]),
+                        };
+                        let response_text =
+                            serde_json::to_string(&data).expect("Bad Serialization of Type");
```

Comment:
I think this is one of those cases where we have unrecoverable errors. I looked at it like we are the ones doing the serialization and at this point the serialization should always pass from our side since we are in charge of the implementation

Reply:
yeah, that might be correct, but I don't see the negative in handling to avoid crashing or to future proof if something else is changed